specs/interop/shared-lockbox-upgrade.md

```diff
@@ -0,0 +1,93 @@
+
+**Migrate ETH liquidity from `OptimismPortal` to `SharedLockbox`**
+
+The ETH will be transferred from the `OptimismPortal` to the `SharedLockbox` using an intermediate contract.
+This contract functions similarly to the `StorageSetter`, being updated prior to the real implementation.
+Its sole purpose is to transfer the ETH balance.
+This approach eliminates the need for additional code to move the liquidity to the lockbox later.
+
+**Set `SuperchainConfig` as dependency manager on `SystemConfig`**
+
+The `SystemConfig` manages the dependency updates from L1 to L2 by making a deposit through the portal.
+It uses access control, allowing only a designated `dependencyManager` to call it.
+The `SuperchainConfig` must be set as the `dependencyManager` since it will be the contract
+responsible for handling the op-governed dependency set. This step is crucial before adding a
+chain to the dependency set, as it is necessary for keeping the dependency set synchronized
+between L1 and L2.
+
+There are two alternatives to accomplish this:
+
+- Upgrade the code to `SystemConfigInterop` and call `initialize` with the dependency manager address
+- Add a setter function for the dependency manager
+
+**Add the chain to the op-governed dependency set**
+
+The `SuperchainConfig` contract will be responsible for storing and managing the dependency set.
+Its `addChain` function will add a chain to the dependency set and call the `SystemConfig` of each chain
+to keep them in sync.
+It will also whitelist the corresponding `OptimismPortal`, enabling it to lock and unlock ETH from the `SharedLockbox`.
+Once this process is complete, the system will be ready to process deposits and withdrawals.
+
+## Batch transaction process
+
+The most efficient approach is to handle the entire migration process in a single batched transaction.
+This transaction will consist of:
+
+1. Call `upgradeAndCall` in the `ProxyAdmin` for the `SystemConfig`
+   - Updating provisionally to the `StorageSetter` to zero out the initialized slot.
+2. Call `upgradeAndCall` in the `ProxyAdmin` for the `SystemConfig`
+   - Calling `initialize` with the `SuperchainConfig` address as the dependency manager
+3. Call `addChain` in the `SuperchainConfig`
+   - Sending chain ID + system config address
+4. Call `upgradeAndCall` in the `ProxyAdmin` for the `OptimismPortal`
+   - Update provisionally to the `LiquidityMigrator` to transfer the whole ETH balance to the `SharedLockbox` in this call.
+5. Call `upgrade` in the `ProxyAdmin` for the `OptimismPortal`
+   - The `SharedLockbox` address is set as immutable in the new implementation
+
+The L1 ProxyAdmin owner will execute this transaction. As the entity responsible for updating contracts,
+it has the authority to perform the first two steps.
+For the third step, the L1PAO has to be set as authorized for adding a chain to the op-governed dependency set
```

Comment:
what is L1PAO? if it has not been defined before, use the full name, unless that is the full name